jvm/src/main/kotlin/app/cash/trifle/Certificate.kt

```diff
@@ -84,6 +124,37 @@ data class Certificate internal constructor(
   companion object {
     internal const val CERTIFICATE_VERSION: Int = 0
 
+    class VerifyResult constructor(val reason: Reason, val msg: String) {
```

Comment:
just use a data class instead which gives you the hash code and equality for free.

Reply:
msg can also be assigned to a default value of an emptyString here.